system/lib/pthread/library_pthread.c

```diff
@@ -664,6 +664,10 @@ int emscripten_dispatch_to_thread_async_(pthread_t target_thread,
   return ret;
 }
 
+int _emscripten_thread_is_valid(pthread_t thread) {
```

Comment:
It would be helpful to say what this function is for. When I first read it I was confused and thought it was checking that `thread` was the current thread.
```suggestion
// Checks thread structural invariants.
int _emscripten_thread_is_valid(pthread_t thread) {
```

Reply:
Add an longer comment to make it clear why we even have this function.